R/Figures.R

```diff
@@ -82,13 +82,104 @@ TADA_boxplot <- function(filtered.data) {
       yaxis = list(title = y_label, titlefont = list(size = 16, family = "Arial"), tickfont = list(size = 16, family = "Arial"),
                    hoverformat = ',.4r', linecolor = "black", rangemode = 'tozero', 
                    showgrid = FALSE, tickcolor= "black"),
-      hoverlabel=list(bgcolor="white")
+      hoverlabel=list(bgcolor="white"),
+      title = paste0("Boxplot of ", char, " (", unit, ")"), 
+      plot_bgcolor = "#e5ecf6"
     ) %>% 
     plotly::config(displayModeBar = FALSE)
   
   return(base_boxplot)
 }
 
+#' Create Histogram
+#' 
+#' @param filtered.data TADA data frame containing the data downloaded from the WQP, where
+#' each row represents a unique data record. Data frame must include the columns
+#' 'TADA.ComparableDataIdentifier", 'TADA.ResultMeasureValue', and
+#' 'TADA.ResultMeasure.MeasureUnitCode' to run this function. These columns can 
+#' be added to the data frame by running the function HarmonizeData(). The data frame
+#' must be filtered down to a single comparable data identifier to run this function.
+#' 
+#' @return A plotly histogram figure showing the distribution of sample values 
+#' for the given comparable data identifier.
+#' 
+#' @export
+#' 
+#' @examples
+#' # Load example dataset:
+#' data("TADAProfileClean18_TNonly")
+#' # TADAProfileClean18_TNonly dataframe is clean, harmonized, and filtered
+#' # down to one Comparable Data Identifier
+#' 
+#' # Create histogram:
+#' TADA_hist(TADAProfileClean18_TNonly)
+
+TADA_hist <- function(filtered.data) {
+  # check .data is data.frame
+  checkType(filtered.data, "data.frame", "Input object")
+  # check .data has required columns
+  checkColumns(filtered.data, c("TADA.ComparableDataIdentifier", "TADA.ResultMeasureValue", "TADA.ResultMeasure.MeasureUnitCode"))
+  # check TADA.ComparableDataIdentifier column is filtered to one identifier
+  if (length(unique(filtered.data$TADA.ComparableDataIdentifier)) > 1) {
+    warning("Histogram function cannot run with more than 1 comparable data identifier. Please filter dataframe and rerun function.")
+  }
+  
+  # execute function after checks have passed
+  
+  # units
+  unit <- unique(filtered.data$TADA.ResultMeasure.MeasureUnitCode)
+  char <- unique(filtered.data$TADA.CharacteristicName)
+  x_label <- paste0("Level of ", char, " (", unit, ")")
```

Comment:
Suggest getting rid of "Level of ", since the x-axis is (usually) the concentration of the characteristic. I think you could just show char (unit).